### ISSUE TITLE
[Analytics accuracy] Timezone differences

### ISSUE THREAD

**Issue by atorresveiga** [woocommerce/woocommerce-android#9887]
It is very common among merchants to have different timezone settings in their stores vs. their devices. This issue could lead to inaccurate stats when analyzing different periods because the data from the app will display different information than the web if they use different timezones. This task is about fixing this issue and using a single truth source. 

pe5pgL-3HL-p2

**Comment by shiki** [comment]
@atorresveiga Thanks for creating this issue.

**Comment by shiki** [comment]
@atorresveiga Another message. 😁

**Comment by jaclync** [comment]
Just wanted to confirm the expected changes for this task: we're going to use the store time zone instead of the device time zone for the analytics because of data we've observed in Tracks. I plan to work on this tomorrow.

It's also interesting that iOS has a lot more unique events than Android.

**Comment by atorresveiga** [comment]
> we're going to use the store time zone instead of the device time zone for the analytics because of data we've observed in Tracks. I plan to work on this tomorrow.

Yes Jaclyn, we should use the store time zone instead of the device and follow the same behavior as the web.

**Comment by jaclync** [comment]
Hi @atorresveiga 👋 Sharing what I've tested in iOS: I tested stores in different time zones, one UTC, one in a different city, and one in the same time zone, and the stats are already showing in the store time zone. The start/end date parameters in the stats API requests don't have a time zone, so the data follow the store time zone. The obvious issue I noticed was pe5pgL-49w-p2 which makes more sense to update on the API side.

I also tried looking at the analytics feedback as in pe5pgL-3HL-p2#increase-the-accuracy-of-analytics, but all the feedback was on Android 🤔 Maybe we didn't launch the survey in iOS for analytics, I couldn't find any relevant survey URLs in iOS.

**Comment by atorresveiga** [comment]
> Hi @atorresveiga 👋 Sharing what I've tested in iOS: I tested stores in different time zones, one UTC, one in a different city, and one in the same time zone, and the stats are already showing in the store time zone.

Here is the test I'm using to reproduce the inconsistency: My store is in the US, and I set the device timezone to Australia. Every day, I create a new order and check the stats. Because in the middle of my day, it is already 1 am in Australia (the next day), when I check stats like Today, Yesterday, This Week (Week to Date), etc, the values differ. The Today selection on my device shows the information for Dec 15, for example, while looking at the same stats on the web displays the data for Dec 14.

> Maybe we didn't launch the survey in iOS for analytics

I believe this is the case ☝️ I created the survey for Android during a Hack Week, but I don't remember if there was an iOS counterpart; I believe not.

**Comment by jaclync** [comment]
> Here is the test I'm using to reproduce the inconsistency: My store is in the US, and I set the device timezone to Australia. Every day, I create a new order and check the stats. Because in the middle of my day, it is already 1 am in Australia (the next day), when I check stats like Today, Yesterday, This Week (Week to Date), etc, the values differ. The Today selection on my device shows the information for Dec 15, for example, while looking at the same stats on the web displays the data for Dec 14.

I see what you meant for this issue now! Opened a discussion in pe5pgL-4bG-p2.

**Comment by toupper** [comment]
@atorresveiga I see we have merged two PRs for this, do we need to do something else, or can we close it? In case there're still pending tasks, what are these? Thanks in advance for your answers!

**Comment by atorresveiga** [comment]
> In case there are still pending tasks, what are these?  In case there are still pending tasks, what are these? Thanks in advance for your answers!

@toupper This refactor was thought to be done progressively until we used the site timezone for every date in the app. Creating issues for every date we use in the app could be overkill, so we decided to take this slow and create issues whenever we identify an inconsistency of a date with another screen (for example, stats and order list). So far, some dates in the order details screen still need to use this site's timezone approach (if I recall correctly, order notes and reviews).

**Comment by Ecarrion** [comment]
@atorresveiga is this still in the plan or should we close it / remove it from our backlog?

**Comment by atorresveiga** [comment]
@Ecarrion I believe some areas still need to be improved, like filters, for example. WDYT @toupper?

**Comment by toupper** [comment]
> @Ecarrion I believe some areas still need to be improved, like filters, for example. WDYT @toupper?

Yes, let's leave it in Inbox 👍

**Comment by atorresveiga** [comment]
The app documentation also mentioned the expected behavior:
https://woocommerce.com/document/how-timezones-work-in-the-woo-mobile-app/

> For both Android and iOS, the Woo Mobile app uses the timezone set on your site’s dashboard (WP-Admin 👉 Settings 👉 General 👉 Timezone) when it’s available, and if it’s not, it will default to the device’s timezone.

cc. @toupper

**Comment by malinajirka** [comment]
Automated test/comment: I checked this during the WooCommerce Android GitHub bug audit. The original Today/Week-to-Date mismatch did not reproduce on the staging site after setting the site timezone to America/Los_Angeles and the emulator timezone to Australia/Sydney, but the issue still appears relevant for the narrower remaining scope called out in the GitHub comments: analytics filters/custom ranges. Current code still uses Calendar.getInstance(), ZoneId.systemDefault(), and Date(...) picker timestamps in those paths.

I migrated the remaining Android work to Linear for triage: https://linear.app/a8c/issue/WOOMOB-3270/android-analytics-filters-still-have-device-timezone-date-paths

Closing this GitHub issue in favor of Linear.
